### PR TITLE
boilerplate: Set up initial Go repo

### PR DESCRIPTION
**Pull request by juliaogris**
Set up initial Go repo with top level `main.go` and test files. Add a
typical Makefile with `build`, `install`, `test`, `run`,
`check-coverage`, `lint` and  `help` targets.

Set up CI configuration for GitHub actions. Add automatic Semver tagging
and GitHub release creation steps to workflow for merge to master.

Add high-level docs - hopefully they make sense.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,8 @@
+module github.com/juliaogris/gurl
+
+go 1.16
```
Comment (camh-) [inline go.mod:3]: Does this work, given that 1.16 is not out yet?

File: .github/workflows/cicd.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        tag_name: ${{ steps.bump_version.outputs.tag }}
+        release_name: Release ${{ steps.bump_version.outputs.tag }}
+        body_path: /tmp/commit-msg
+    # Get the latest module version so pkg.go.dev updates
+    - uses: WillAbides/setup-go-faster@v1
+      with:
+        go-version: 1.16rc1
+    - name: Update pkg.go.dev
+      working-directory: /tmp
+      run: go get github.com/juliaogris/gurl@${{ steps.bump_version.outputs.tag }}
```
Comment (camh-) [inline .github/workflows/cicd.yaml:62]: It looks like we might be able to do this now without Go installed: https://github.com/golang/go/issues/37106#issuecomment-690419151 - do you want to try?

File: README.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+[![Slack chat](https://img.shields.io/badge/slack-gophers-795679?logo=slack)](https://gophers.slack.com/messages/grpc)
+
+`gurl` is like `curl` for [gRPC](https://grpc.io/).
+
+This project is a shim for the great [`grpcurl`](https://github.com/fullstorydev/grpcurl).
+`gurl` uses slightly extended CLI options and an HTTP extension to work with [gRPC-Gateway](https://github.com/grpc-ecosystem/grpc-gateway).
+
+### Development
+
+- Pre-requisites: [go 1.16](https://golang.org), [golangci-lint](https://github.com/golangci/golangci-lint/releases/tag/v1.36.0), GNU make
```
Comment (camh-) [inline README.md:14]: Why is 1.16 a prerequisite? Will you be using embedding or `io.fs`?

File: go.mod (hunk 1 above)
Comment (juliaogris) [reply go.mod:3]: Not sure. I ran `go mod init` and that's what it did.
Go 1.16 should be out within the next couple of weeks so I think that's acceptable....

File: README.md (hunk 3 above)
Comment (juliaogris) [reply README.md:14]: No but it will be out so soon that I would like to use it.

File: .github/workflows/cicd.yaml (hunk 2 above)
Comment (camh-) [reply .github/workflows/cicd.yaml:62]: It looks like there are still issues around updating pkg.go.dev - perhaps leave this as is for now.

**Review by camh- — APPROVED** [review]
:herb: LGTM

File: .github/workflows/cicd.yaml (hunk 2 above)
Comment (juliaogris) [reply .github/workflows/cicd.yaml:62]: yeah, issue isn't fully resolved yet so let's leave as is.

**Pull request merged**